components/ai_chat/resources/page/components/attachment_item/index.tsx

```diff
@@ -139,3 +119,62 @@ export function AttachmentPageItem(props: { title: string, url: string, remove?:
     </>}
     remove={props.remove} />
 }
+
+export function AttachmentUploadItem(props: {
+  uploadedFiles: Mojom.UploadedFile[]
+  remove?: (index: number) => void
+}) {
+  return (
+    <>
+      {props.uploadedFiles.map((file, index) => {
+        const isImage = file.type === Mojom.UploadedFileType.kImage ||
+                       file.type === Mojom.UploadedFileType.kScreenshot
+        const isPdf = file.type === Mojom.UploadedFileType.kPdf
+
+        if (isImage) {
+          const dataUrl = React.useMemo(() => {
+            const blob = new Blob([new Uint8Array(file.data)], {
+              type: 'image/*'
+            })
+            return URL.createObjectURL(blob)
+          }, [file])
+
+          const filesize = React.useMemo(() => {
+            return formatFileSize(Number(file.filesize))
+          }, [file.filesize])
+
+          return (
+            <AttachmentItem
+              key={file.filename}
+              icon={
+                <img
+                  className={styles.image}
+                  src={dataUrl}
+                />
+              }
+              title={file.filename}
+              subtitle={filesize}
+              remove={props.remove ? () => props.remove!(index) : undefined}
+            />
+          )
+        } else if (isPdf) {
+          const filesize = React.useMemo(() => {
+            return formatFileSize(Number(file.filesize))
+          }, [file.filesize])
+
+          return (
+            <AttachmentItem
+              key={file.filename}
```

Comment:
hey, not 100% sure but `filename` can probably be the same on multiple attachments (say, if the user uploads the same one again).

Might need to use the `index` here :/

Reply:
used filename and index as key in 53509362ade970cc99956e5006a8665150ca4136